### PR TITLE
Updates to Platform.sh deploy instructions

### PR DESCRIPTION
**Pull request by pjcdawkins**
* Update example PHP version from 5.6 to 7.1
* Files in `.platform` are now required
* Endpoint name `php` is better written as `http`
* Build flavor `symfony` is better written as `composer`
* Rename database from `mysql` to `mydatabase` to separate it clearly from the service type
* The `web` section should now use `locations`
* Use SSL by default (certificate provisioning and 'http://' redirects are automatic)

There are probably additional improvements that can be made, referring to the Platform.sh [documentation](https://docs.platform.sh) and to the work that has been done in https://github.com/platformsh/platformsh-example-symfony/

### CODE REVIEW

File: deployment/platformsh.rst [hunk 1]
```diff
@@ -39,23 +39,25 @@ Platform.sh how to deploy your application (read more about
     name: myphpproject
     
     # The type of the application to build.
-    type: php:5.6
+    type: php:7.1
     build:
-      flavor: symfony
+      flavor: composer
```
Comment (OskarStark) [inline deployment/platformsh.rst]: why do you change this? `symfony` is correct

File: deployment/platformsh.rst [hunk 2]
```diff
@@ -73,21 +75,21 @@ Platform.sh how to deploy your application (read more about
...
 
 .. code-block:: yaml
 
     # .platform/routes.yaml
-    "http://{default}/":
+    'https://{default}/':
         type: upstream
         # the first part should be your project name
-        upstream: 'myphpproject:php'
+        upstream: 'myphpproject:http'
```
Comment (OskarStark) [inline deployment/platformsh.rst]: should be:

```yaml
'https://{default}/':
  type: upstream
  upstream: "myphpproject:http"
'https://www.{default}/':
  type: redirect
  to: "https://{default}/"
```

**Review by OskarStark — CHANGES_REQUESTED** [review]

**Comment by OskarStark** [comment]
Refs #8483

File: deployment/platformsh.rst (hunk 1 above)
Comment (javiereguiluz) [reply deployment/platformsh.rst]: I read somewhere that `composer` is the new and recommended type for all PHP. Their Symfony template doc use that too: https://docs.platform.sh/frameworks/symfony.html

Comment (OskarStark) [reply deployment/platformsh.rst]: good to know, thank you @javiereguiluz 👍

File: deployment/platformsh.rst [hunk 3]
```diff
@@ -73,21 +75,21 @@ Platform.sh how to deploy your application (read more about
         deploy: |
           bin/console --env=prod cache:clear
 
-For best practices, you should also add a ``.platform`` folder at the root of
+You also need add a ``.platform`` folder at the root of
```
Comment (HeahDude) [inline deployment/platformsh.rst]: to add

**Review by HeahDude — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@pjcdawkins thanks for this contribution, but sadly we must close this pull request without merging it. Why? Because in #8721 we decided to remove all these articles explaining how to deploy to third-party services. We now simply link to the articles published by those services. Thanks for understanding it.

**Pull request closed without merging**